### ISSUE TITLE
Error using Custom Auth with GRPC (api-gateway)

### ISSUE THREAD

**Issue by galbino** [GoogleCloudPlatform/esp-v2#682]
Hi, I'm trying to use the API Gateway with custom Authentication in my GRPC (golang) project, but, after configuring everything, I keep getting the following error:
`{
    "code": 401,
    "message": "Jwks remote fetch is failed"
}`
My YAML config file is like this:
`
type: google.api.Service
config_version: 3
name: "*.apigateway.<project-id>.cloud.goog"
title: <title>
apis:
  - name: <name>
authentication:
  providers:
  - id: <auth_id>
    issuer: <issuer>
    jwks_uri: https://storage.googleapis.com/test-jwks-dev/jwks.json
    audiences: <audience>
    jwt_locations:
    - header: "authorization"
      value_prefix: "Bearer "
  rules:
  - selector: "<method-test>"
    requirements:
    - provider_id: <auth_id>
usage:
  rules:
    - selector: "*"
      allow_unregistered_calls: true
backend:
  rules:
    - selector: "*"
      address: grpcs://<cloud_run_address>
`
I've looked everywhere and tried everything, but its still not working.
My JWKS URI seems fine, I even tested with a validator, but it doesnt look like its going to work. The JWT seems fine aswell.

I'm running the GRPC server with golang on cloud run.

**Comment by qiwzhang** [comment]
Could you turn on debug mode to get the debug info from the proxy?  I am not sure how to do that in Api-gateway.

**Comment by galbino** [comment]
I'm not sure how to do that either.
Actually I dont think thats even possible. The closest I can do is check the API Gateway logs and this is the error message from that request:
```
{
api_key_state: "NOT CHECKED"
api_method: "<method>"
api_name: "<name>"
api_version: "v1"
http_status_code: 401
location: "us-central1"
log_message: "<service> is called"
producer_project_id: "-"
response_code_detail: "jwt_authn_access_denied{Jwks_remote_fetch_is_failed}"
service_agent: "ESPv2/2.35.0"
service_config_id: "back-to-req-7-2314vyggfhzxh"
timestamp: 1649708711.3428657
}
```
Is that remotely close to what you need?

**Comment by qiwzhang** [comment]
No, we need Envoy debug log. You need to get help from api-gateway team.  This is ESPv2 team.

**Comment by galbino** [comment]
Oh, ok.
Could you point me to how I can communicate with them?
I only found the repository belonging to the esp-v2, and, since Apigateway uses it, i thought you guys could help me.

**Comment by qiwzhang** [comment]
I have pinged api-gateway team internally.

**Comment by qiwzhang** [comment]
Responded.    Asked you to report issue [here](https://b.corp.google.com/issues/new?component=961566&template=1508911&pli=1)

**Comment by galbino** [comment]
I dont think I can open an issue there.
![image](https://user-images.githubusercontent.com/29392550/162831052-a3cefb9a-182b-4199-9804-0bac0e6c5f68.png)

**Comment by qiwzhang** [comment]
Sorry wrong link.  Should use this [one](https://issuetracker.google.com/issues/new?component=961566&template=0 )

**Comment by galbino** [comment]
Ok!
I'm gonna create there so i'll close this issue.
Thanks for your assistance.
